lib/yabeda/datadog/logging.rb

```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+
+require "logger"
+require "singleton"
+
+module Yabeda
+  module Datadog
+    # = Perform loging
+    class Logging
+      include Singleton
+
+      def initialize
+        @logger = Logger.new(STDOUT)
+      end
+
+      def log_request(metric)
+        info "Sending #{metric.name} metric"
+        response = yield
+        info "Response on #{metric.name}: #{handle_response(response)}"
+        response
+      end
+
+      def warn(message)
+        @logger.warn message
+      end
+
+      def info(message)
+        @logger.info message
+      end
+
+      def debug(message)
+        @logger.debug message
+      end
+
+      def fatal(message)
+        @logger.fatal message
+      end
+
+      def error(message)
+        @logger.error message
+      end
+
+      def level=(level)
+        @logger.level = level
+      end
+
+      private
+
+      def handle_response(response)
```

Comment:
Sorry to not be clear enough. I think that `#handle_response` do not belong to Logging module either, but belong to something else.

Reply:
I extract all response related logic from logging class to handle response class.